#!/usr/bin/env python
PACKAGE = "teleop_pwm_override"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()

gen.add("override_enable", bool_t, 0, "Send overrides", False)
gen.add("failsafe_enable", bool_t, 0, "Send PWM low for all channels", True)

gen.add("chan1_enable", bool_t, 0, "Enable channel 1 override", False)
gen.add("chan1_raw",     int_t, 0,        "Channel 1 override",  1000, 1000, 2000)
gen.add("chan2_enable", bool_t, 0, "Enable channel 2 override", False)
gen.add("chan2_raw",     int_t, 0,        "Channel 2 override",  1000, 1000, 2000)
gen.add("chan3_enable", bool_t, 0, "Enable channel 3 override", False)
gen.add("chan3_raw",     int_t, 0,        "Channel 3 override",  1000, 1000, 2000)
gen.add("chan4_enable", bool_t, 0, "Enable channel 4 override", False)
gen.add("chan4_raw",     int_t, 0,        "Channel 4 override",  1000, 1000, 2000)
gen.add("chan5_enable", bool_t, 0, "Enable channel 5 override", False)
gen.add("chan5_raw",     int_t, 0,        "Channel 5 override",  1000, 1000, 2000)
gen.add("chan6_enable", bool_t, 0, "Enable channel 6 override", False)
gen.add("chan6_raw",     int_t, 0,        "Channel 6 override",  1000, 1000, 2000)
gen.add("chan7_enable", bool_t, 0, "Enable channel 7 override", False)
gen.add("chan7_raw",     int_t, 0,        "Channel 7 override",  1000, 1000, 2000)
gen.add("chan8_enable", bool_t, 0, "Enable channel 8 override", False)
gen.add("chan8_raw",     int_t, 0,        "Channel 8 override",  1000, 1000, 2000)

exit(gen.generate(PACKAGE, "teleop_pwm_override_node", "PWMOverride"))
